src/visudo/mod.rs

```diff
@@ -88,15 +88,15 @@ fn run_visudo(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()
         }
     })?;
 
-    if perms || file_arg.is_none() {
-        sudoers_file.set_permissions(Permissions::from_mode(0o440))?;
-    }
+    let result: io::Result<()> = (|| {
```

Comment:
as an alternative to this closure that gets called immediately ("pseudo try-block"), I think you could create a newtype over a locked file that releases the lock when it's dropped. so long we don't call `exit` from here that should release the lock even when an error occurs. 

that being said, according to the [glibc manual](https://www.gnu.org/software/libc/manual/html_node/File-Locks.html), the locks associated to a file are released when the file is closed and I think that should happen even if the process call `exit` because the OS should release all the resources associated to the process.